src/middlewares/Fetch.ts

```diff
@@ -1,22 +1,26 @@
 import {combineUrlWithBaseUrl, combineUrlWithQueryParameters} from "../internal/formatting";
 import {IMiddleware} from "../Stack";
+import {IFetchRequestCacheOptions} from "./Cache";
 
 export type FetchHeaders = {
     [key: string]: string;
+    forEach?: any;
 };
 
 export type FetchResponse<T> = {
     new(body: any, init: any): FetchResponse<T>;
     body: string;
     bodyUsed: boolean;
-    headers?: FetchHeaders;
+    headers: FetchHeaders;
     ok: boolean;
     status: number;
     statusText: string;
     type: string;
     url: string;
+    cache?: { used: boolean; timestamp: number; }; // signifies that response has been reconstructed from cache
```

Comment:
How about we add an age property here as well? Or is this adaptive to official specs? :) If not, we could also have an `IFetchResponseCacheOptions` similar to `IFetchRequestCacheOptions`.